File: src/main/java/de/flapdoodle/generics/uniontypes/j/components/DateField.java
package de.flapdoodle.generics.uniontypes.j.components;

import de.flapdoodle.generics.uniontypes.j.framework.Component;
import de.flapdoodle.generics.uniontypes.j.framework.HasValidation;
import de.flapdoodle.generics.uniontypes.j.framework.HasValue;

import java.time.LocalDate;

public class DateField extends Component implements HasValue<LocalDate>, HasValidation {
  private LocalDate value = null;
  private boolean invalid= false;

  @Override
  public LocalDate getValue() {
    return value;
  }

  @Override
  public void setValue(LocalDate value) {
    this.value = value;
  }

  @Override
  public void setInvalid(Boolean flag) {
    this.invalid = flag;
  }
}

File: src/main/kotlin/de/flapdoodle/generics/uniontypes/k/components/DateField.kt
package de.flapdoodle.generics.uniontypes.k.components

import de.flapdoodle.generics.uniontypes.k.framework.Component
import de.flapdoodle.generics.uniontypes.k.framework.HasValidation
import de.flapdoodle.generics.uniontypes.k.framework.HasValue
import java.time.LocalDate

class DateField : Component(), HasValue<LocalDate>, HasValidation {
  private var value: LocalDate? = null
  private var invalid: Boolean = false

  override fun getValue(): LocalDate? = value

  override fun setValue(value: LocalDate?) {
    this.value = value
  }

  override fun setInvalid(flag: Boolean) {
    this.invalid=flag
  }
}